python/lsst/qa/explorer/classifyObjects.py

```diff
@@ -0,0 +1,495 @@
+
+
+class StarGalaxyClassifierConfig(pexConfig.Config):
+    """Config for star galaxy classifier
+
+    Notes
+    -----
+    The version name is specifed by user in the mkClassifier task and is used to read in the correct
+    classifiers.
+    """
+    version = pexConfig.Field(
+        doc="Version information",
+        dtype=str,
+        default="HSC-COSMOS-20180528",
+    )
+
+    seeingColName = pexConfig.Field(
+        doc="Column name for seeing columns used in addSeeing",
+        dtype=str,
+        default="base_SdssShape_psf"
+    )
+
+    psfColName = pexConfig.Field(
+        doc="Column name for PSF flux columns used in addMagnitudes",
+        dtype=str,
+        default="base_PsfFlux_flux"
+    )
+
+    modelColName = pexConfig.Field(
+        doc="Column name for model flux columns used in addSNFlux and addMagnitudes",
+        dtype=str,
+        default="modelfit_CModel_flux"
+    )
+
+
+class StarGalaxyClassifierTask(Task):
+    """Add classification (star or galaxy) and associated flag columns to a parquet table.
+
+    Classify each object in the catalog into either a star or a galaxy using two different methods, one that
+    uses purely morphological information and one that uses both morphological and color information. Add an
+    associated flag column for each type of classifier that is set if any NaNs were present in the input data.
+    This is done for the patch specified in the ``DataRef``. To apply the classifiers additional columns are
+    needed beyond those in the standard ``deepCoadd_obj``, add these into a new index (featuresSGSep) in the
+    parquet table. The classification columns are ``starProbColors``, ``starProbMorph``,
+    ``starProbColorsFlag`` and ``starProbMorphFlag`` these contain the probabilites of each object being a
+    star according to both classifiers and a flag for each. Details of the new columns are summarised below.
+
+    - starProbColors: probability the object is a star, based on morphology and color information, fraction
+                      where 1 = star
+    - starProbColorsFlag: set if starProbColors cannot be determined, e.g. if any of the columns needed by the
+                      classifier contain NaNs.
+    - starProbMorph: probability the object is a star, based on morphology information only, fraction where
+                     1 = star
+    - starProbMorphFlag: set if starProbMorph cannot be determined, e.g. if any of the columns needed by the
+                         classifier contain NaNs.
+
+    `StarGalaxyClassifierTask` has five functions that add the extra columns needed to a new index in the
+    parquet table.
+
+    addMagnitudes
+        Use the information in the ``deepCoadd_calexp_calib`` to add CModel magntidue and error columns for
+        each band to the new index in the parquet table
+    addSeeing
+        Add a column of the geometric mean of the ``base_SdssShape_psf`` matrix to the new index in the
+        parquet table
```

Comment:
Please update `base_SdssShape_psf` and the other hard-coded column names to the new config parameters you so kindly added. I suggest a global search for each one. In this case I suggest being more generic, e.g. "Add a column of the geometric mean of the seeing to the (?a?) new index..."